From 858bdd51a8e44ce7038559c704fb94a0af612588 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 17 Sep 2015 15:44:29 -0500
Subject: [PATCH 47/97] apply backport patch
 collateral-evolutions/network/0029-struct_hid_ll_driver-extensions/net_bluetooth_hidp_core.patch

---
 backports/net/bluetooth/hidp/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/backports/net/bluetooth/hidp/core.c b/backports/net/bluetooth/hidp/core.c
index 9070dfd..517c8f6 100644
--- a/backports/net/bluetooth/hidp/core.c
+++ b/backports/net/bluetooth/hidp/core.c
@@ -224,6 +224,7 @@ static void hidp_input_report(struct hidp_session *session, struct sk_buff *skb)
 	input_sync(dev);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,15,0)
 static int hidp_get_raw_report(struct hid_device *hid,
 		unsigned char report_number,
 		unsigned char *data, size_t count,
@@ -396,6 +397,7 @@ static int hidp_raw_request(struct hid_device *hid, unsigned char reportnum,
 		return -EIO;
 	}
 }
+#endif
 
 static void hidp_idle_timeout(unsigned long arg)
 {
@@ -725,8 +727,10 @@ static struct hid_ll_driver hidp_hid_driver = {
 	.stop = hidp_stop,
 	.open  = hidp_open,
 	.close = hidp_close,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,15,0)
 	.raw_request = hidp_raw_request,
 	.output_report = hidp_output_report,
+#endif
 };
 
 /* This function sets up the hid device. It does not add it
-- 
2.5.1

